app/controllers/concerns/discovery.rb

```diff
@@ -1,18 +1,47 @@
+require "velum/salt_minion"
+require "velum/salt_api"
 # Discovery implements the discovery method that is shared both when
 # bootstrapping and when showing the dashboard.
 module Discovery
   extend ActiveSupport::Concern
+  include Velum::SaltApi
 
   # Responds with either an HTML or JSON version of the available minions.
   def discovery
     @assigned_minions = Minion.assigned_role
     @unassigned_minions = Minion.unassigned_role
 
+    tx_update_reboot_needed = Hash.new
+    tx_update_failed = Hash.new
+
+    minions = perform_request(endpoint: "/minions", method: "get")
+
+    #logger.info JSON.parse(minions.body)["return"]
+
+    ret = JSON.parse(minions.body)["return"]
+
+    for minion in ret do
+      minion.each do | key, value |
+        if value["tx_update_reboot_needed"]
```

Comment:
Also, renaming the arguments to `minion_id`, `grains` would be great, so it's something more intuitive.